scalafmt-core/shared/src/main/scala/org/scalafmt/config/RewriteScala3Settings.scala

```diff
@@ -56,4 +60,26 @@ object RewriteScala3Settings {
 
   }
 
+  case class ConvertToNewSyntax(
+      ifCond: Boolean = true,
+      whileCond: Boolean = true,
+      forEnum: Boolean = true,
+      varargStar: Boolean = true,
+      importStar: Boolean = true,
+      importAs: Boolean = true,
+      typeWildcardQuestionMark: Boolean = true,
+      typePlaceholderUnderscore: Boolean = true
```

Comment:
```suggestion
      deprecatedSyntax = true
```
maybe?  This would be controlled by the dialect to see if the new syntax is allowed.

Otherwise if only one thing is set then the user will be quite confused. The separate non optional braces (traditional braced if etc.) will probably not be deprecated ever,.